src/data/polynomial/eval.lean

```diff
@@ -538,9 +538,15 @@ end
 @[simp] protected lemma map_mul : (p * q).map f = p.map f * q.map f :=
 by { rw [map, eval₂_mul_noncomm], exact λ k, (commute_X _).symm }
 
-@[simp] lemma map_smul (r : R) : (r • p).map f = f r • p.map f :=
+@[simp] protected lemma map_smul (r : R) : (r • p).map f = f r • p.map f :=
 by rw [map, eval₂_smul, ring_hom.comp_apply, C_mul']
 
+@[simp] protected lemma map_bit0 : (bit0 p).map f = bit0 (p.map f) :=
+by simp only [bit0, polynomial.map_add]
+
+@[simp] protected lemma map_bit1 : (bit1 p).map f = bit1 (p.map f) :=
+by simp only [bit1, polynomial.map_add, polynomial.map_bit0, polynomial.map_one]
```

Comment:
I'd argue you should prove these after `map_ring_hom`, since then the proofs are absolutely trivial and follow the same patter as the `map_nat_cast` that's already there.